File: backend/dataset/levelOrder/levelorder_16.py
from queue import Queue 

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if root is None:
            return []
        
        q = Queue()
        q.put(root)
        res = []
        while (q.qsize()):
            nextLevel = Queue()
            levelArr = []
            
            # For all nodes on the same level, add child nodes from left to right
            while (q.qsize()):
                front = q.get()
                levelArr.append(front.val)
                if front.left != None:
                    nextLevel.put(front.left)
                if front.right != None:
                    nextLevel.put(front.right)
                    
            # By this point, q has exhausted all nodes on that level
            q = nextLevel
            
            # Add the level array to the result array
            res.append(levelArr)
        
        return res

File: backend/CDataset/levelOrder/levelOrder_16.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
       queue<TreeNode*>q;
       vector<vector<int>>ans;
        if(root==NULL) return ans;
        q.push(root);
        while(q.empty()!=true)
        {   vector<int>v;
            int n=q.size();
            for(int i=0;i<n;i++)
            {   TreeNode* t=q.front();q.pop();
                v.push_back(t->val);
                if(t->left) q.push(t->left);
                if(t->right) q.push(t->right);
            }
            ans.push_back(v);
        }
        return ans;
    }
};
``